core/main/src/bootstrap/extn/start_extn_channel_step.rs

```diff
@@ -82,11 +84,20 @@ impl Bootstep<BootstrapState> for StartExtnChannelsStep {
                 .extn_state
                 .add_extn_status_listener(extn_id.clone(), tx)
             {
-                if let Some(v) = tr.recv().await {
-                    state.extn_state.clear_status_listener(extn_id);
-                    match v {
-                        ExtnStatus::Ready => continue,
-                        _ => return Err(RippleError::BootstrapError),
+                match timeout(Duration::from_millis(10000), tr.recv()).await {
```

Comment:
Is it okay to have such a long waiting period blocking the boot sequence?

Reply:
The extensions can be implemented to

1. Connect to Thunder or underlying API, 
2. Setup for pub-sub connections 

Want to start with a big number and then get values across all devices to come up with a reasonable threshold. 
This is the worst-case scenario in most cases the extensions start less than 300 mSecs with all the loaded functionality in parallel.

The total bootstrap step loading all extensions with device and distributor connections takes less than 300 msecs.
```
2023-08-16-08:28:05.311[DEBUG][ripple_sdk::framework::bootstrap][gateway]->>>Starting Bootstep StartExtnChannelsStep<<<
2023-08-16-08:28:05.807[DEBUG][ripple_sdk::framework::bootstrap][gateway]----Successful Bootstep StartExtnChannelsStep---
```